devito/data/data.py

```diff
@@ -190,7 +202,45 @@ def __str__(self):
     def __getitem__(self, glb_idx, comm_type, gather_rank=None):
         loc_idx = self._index_glb_to_loc(glb_idx)
         is_gather = True if isinstance(gather_rank, int) else False
-        if comm_type is index_by_index or is_gather:
+        if is_gather and comm_type == gather:
```

Comment:
Refresh my mind please -- what does the user need to write so that we end up here?

Reply:
For MPI gathers, the decorator will infer how to gather based on the slice (or lack of) supplied by the user. (That is, `gather` vs `index_by_index` is set by the decorator, not by the user).